packages/react-instantsearch-hooks-dom/src/ui/Pagination.tsx

```diff
@@ -52,26 +96,34 @@ export function Pagination({
           <PaginationItem
             isDisabled={isFirstPage}
             className="ais-Pagination-item--firstPage"
-            aria-label="First"
+            aria-label={translations.ariaFirst}
             href={createURL(firstPageIndex)}
             onClick={() => onNavigate(firstPageIndex)}
           >
-            ‹‹
+            {translations.first}
           </PaginationItem>
         )}
         {showPrevious && (
           <PaginationItem
             isDisabled={isFirstPage}
             className="ais-Pagination-item--previousPage"
-            aria-label="Previous"
+            aria-label={translations.ariaPrevious}
             href={createURL(previousPageIndex)}
             onClick={() => onNavigate(previousPageIndex)}
           >
-            ‹
+            {translations.previous}
           </PaginationItem>
         )}
         {pages.map((page) => {
-          const label = String(page + 1);
+          const label =
+            typeof translations.page === 'string'
+              ? translations.page
+              : translations.page(page);
+
+          const ariaLabel =
+            typeof translations.ariaPage === 'string'
+              ? translations.ariaPage
+              : translations.ariaPage(page);
```

Comment:
the aria label for a dynamic page makes no sense to be a string. As this is for internal use only, we should accept only the function I guess?